app/controllers/works_controller.rb

```diff
@@ -26,7 +39,7 @@ def check_auth
   end
 
   def decorator
-    @decorator = WorkShowDecorator.new(@work)
+    @decorator
```

Comment:
Instead of making this an iVar set in a `before_action`, just make it conditional here and memoized. (And I see there was a bug before, even in the before-change version, that `=` should have been `||=`. 

```ruby
@decorator ||= if something
    WorkShowDecorator.new(@work)
else
    AudioWorkShowDecorator(@work)
end
```

Reply:
Yup -- thanks!